course-materials/lab-instructions/lab-06/lab-06-ugly-charts.Rmd

```diff
@@ -48,35 +48,96 @@ use_git_config(user.name = "Your Name",
+
+- **Role 1:**
+
+  * Change the team name to your actual team name.
+  * Knit, commit, push.  
 
-- **Member 1:** Change the team name to your actual team name, knit, commit, push.  
 `r emo::ji("stop_sign")` Wait for instructions before moving on to the next step.
-- **Member 2:** Change the team name to some other word, knit, commit, push. You should get an error. Pull. Take a look at the document with the merge conflict. Clear the merge conflict by choosing the correct/preferred change. Commit, and push.  
+
+- **Role 2:**
+
+  * Change the team name to some other word.
+  * Knit, commit, push. You should get an error.
+  * Pull. Take a look at the document with the merge conflict.
+  * Clear the merge conflict by choosing the correct/preferred change.
+  * Knit.
+  * **Click the Stage checkbox** for all files in your Git tab. Make sure they
+    all have check marks, not filled-in boxes.
+  * Commit and push.  
+
 `r emo::ji("stop_sign")` Wait for instructions before moving on to the next step.
-- **Member 3:** Add a label to the first code chunk, knit, commit, push. You should get an error. Pull. No merge conflicts should occur. Now push.  
+
+- **Role 3:**
+
+  * Add a label to the first code chunk
+  * Knit, commit, push. You should get an error.
+  * Pull. No merge conflicts should occur, but you should see a message about merging.
+  * Now push.
+
 `r emo::ji("stop_sign")` Wait for instructions before moving on to the next step.
-- **Member 4:** Add a different label to the first code chunk, knit, commit, push. You should get an error. Pull. Take a look at the document with the merge conflict. Clear the merge conflict by choosing the correct/preferred change. Commit, and push.  
+
+- **Role 4:**
+
+  * Add a different label to the first code chunk.
+  * Knit, commit, push. You should get an error.
+  * Pull. Take a look at the document with the merge conflict. Clear the merge conflict by choosing the correct/preferred change. Commit, and push.  
 `r emo::ji("stop_sign")` Wait for instructions before moving on to the next step.  
+
 - **All members:** Pull, and observe the changes in your document.
 
 ### Tips for collaborating via GitHub
 
 - Always pull first before you start working.
-- Commit, and push, often to minimize merge conflicts and/or to make merge conflicts easier to resolve.
-- If you find yourself in a situation that is difficult to resolve, ask questions asap, don't let it linger and get bigger.
+- Resolve a merge conflict (commit and push) *before* continuing your work. Never do
+  new work while resolving a merge conflict.
+- Knit, Commit, and Push often to minimize merge conflicts and/or to make merge conflicts easier to resolve.
+- If you find yourself in a situation that is difficult to resolve, ask questions ASAP. Don't let it linger and get bigger.
+
+(Optional further reading: [Merge conflicts in the classroom](https://github.blog/2018-08-22-merge-conflicts-in-the-classroom/).)
```

Comment:
As much as I love Vanessa's writing on this topic, I hesitate to include this here because the workflow is different than what the curriculum teaches. I find that mixing and matching Terminal instructions with RStudio Git pane instructions confuses students who are new to computing and version control. So I'd suggest taking this out. Unfortunately Happy Git with R also addresses this at a level higher than what's appropriate for this curriculum, IMO.

Reply:
Agreed.